dev/io.openliberty.microprofile.telemetry.2.0.internal/src/io/openliberty/microprofile/telemetry20/internal/http/MPTelemetryHTTPMetricsAdapterImpl.java

```diff
@@ -123,4 +111,35 @@ private Attributes retrieveAttributes(HttpStatAttributes httpStatAttributes) {
         return attributesBuilder.build();
     }
 
+    /*
+     * We can re-use the (histogram) Meter created here.
+     * The Meter is built using the same static values each time.
+     * The instrument that is recorded/updated is distinct for each
+     * http-route/response/method combination (corresponds with resolved attributes).
+     *
+     * However we cannot share it across multiple instances of OpenTelemetry
+     */
+    private DoubleHistogram getHTTPHistogram(OpenTelemetry otelInstance) {
+
+        try {
+            rwl.readLock().lock();
+            if (httpHistogramMap.containsKey(otelInstance)) {
+                return httpHistogramMap.get(otelInstance);
+            }
```

Comment:
Is there any reason to call `containsKey` and then `get` rather than calling `get` directly and check for `null`?

I don't think the `computeIfAbsent` factory lambda can return `null`

Reply:
I think we're safe. If Telemetry is disabled we'll get a no-op objectm and if Telemetry is enabled the chain of methods starting at `build()` cannot return null (it actually ends in a functional interface calling a constructor). 

So the only reason is that `containsKey` is more idiomatic, and since I'm reading the implication that you think we should prefer performence over idom in this class; and can't argue with that given how high traffic this method is, I'll swap to get and checking for null.

Reply:
I would argue that `containsKey` is not idiomatic and is usually to be avoided unless it's expected that the map may contain `null` as a key or if you really just want to check for the presence of a key and don't want the value.